app/module/drivers/sensor/battery/battery_common.c

```diff
@@ -29,15 +29,37 @@ int battery_channel_get(const struct battery_value *value, enum sensor_channel c
     return 0;
 }
 
-uint8_t lithium_ion_mv_to_pct(int16_t bat_mv) {
-    // Simple linear approximation of a battery based off adafruit's discharge graph:
-    // https://learn.adafruit.com/li-ion-and-lipoly-batteries/voltages
+uint8_t lithium_ion_mv_to_pct(int16_t batt_mv) {
+    // Lookup table of slope formulas for calculating remaining battery capacity.
+    // The original values used to calculate slopes come from:
+    //
+    //       https://blog.ampow.com/lipo-voltage-chart/
+    //
+    struct lookup_point {
+        float millivolts;
```

Comment:
The constants in the table are all integers, so we can save a few bytes by storing these fields as `int16_t` instead of 32-bit floats.

Then we can either cast to float when doing the math that uses these numbers, or we can change the algorithm so it works with integers as I suggest below.